src/headers.ts

```diff
@@ -12,17 +17,21 @@ export class Headers {
     if (init) this.extend(init);
   }
 
-  set(headerName: string, value: string | string[]): void {
+  set(headerName: string, value: HeaderValueInput): void {
     this.object[headerName.toLowerCase()] =
-      typeof value === "string" ? value : value.map(String);
+      Array.isArray(value) ? value.map(String) : String(value);
   }
 
-  append(headerName: string, value: string | string[]): void {
+  append(headerName: string, value: HeaderValueInput): void {
     const key = headerName.toLowerCase();
     const prevValue = this.object[key];
     // tslint:disable-next-line
     if (prevValue === undefined) {
-      this.object[key] = typeof value === "string" ? value : value.map(String);
+      if (Array.isArray(value)) {
```

Comment:
Why not re-use this.set(headerName) here?

Reply:
Personally, function overhead and we already have `key` computed. It'd be preferable to pull the functionality into a function independent of either function, but not necessary right now.

Reply:
Fair enough.  :+1: